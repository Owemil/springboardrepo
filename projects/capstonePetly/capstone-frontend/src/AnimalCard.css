.card-img {
    z-index: 97;
    height: 250px;

}

.card-body {
    text-align: center;
}

.card-animal {
    margin-top: 1rem;
    margin-left: 1rem;
}

.card-animal :hover {
    cursor: pointer;
    box-shadow: 1px 1px 6px black;
}


.fav-btn {
    position: absolute;
    z-index: 98;
    background: rgba(255, 255, 255, 0.7);
    border-radius: 50%;
    height: 3em;
    width: 3em;
}

.fav-icon {
    text-align: center;
}