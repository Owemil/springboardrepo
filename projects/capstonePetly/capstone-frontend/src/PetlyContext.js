import React from "react";

const PetlyContext = React.createContext();

export default PetlyContext;