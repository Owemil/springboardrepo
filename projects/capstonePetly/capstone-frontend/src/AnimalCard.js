import React, { useContext } from "react";
import PetlyContext from "./PetlyContext";
import "./AnimalCard.css"
import { Card, CardBody, CardTitle, CardImg, CardText } from "reactstrap";
import Favorite from "./Favorite";

// AnimalCard component takes in animal data from different sources and builds a card based on the data sent through

function AnimalCard({ nearby, search, id, fav }) {
    const { user, getAnimal } = useContext(PetlyContext)

    const name = nearby ? nearby.name.toLowerCase() : search ? search.name.toLowerCase() : fav ? fav.petName : ""

    let favPets = user ? user.favPets.map(animal => animal.petId) : []

    const handleClick = async () => {
        await getAnimal(id)
    }

    if (nearby) {
        return (
            <div className="card-animal">
                <Card
                    style={{
                        width: '18rem'
                    }}
                    onClick={handleClick}
                >
                    {!user ?
                        <Favorite
                            pet={{
                                petId: id,
                                petName: name,
                                petPic: nearby.pictureThumbnailUrl
                            }}
                        />
                        : favPets.includes(+id) ?
                            <Favorite
                                pet={{
                                    petId: id,
                                    petName: name,
                                    petPic: nearby.pictureThumbnailUrl
                                }}
                                currFav={true} />
                            :
                            <Favorite
                                pet={{
                                    petId: id,
                                    petName: name,
                                    petPic: nearby.pictureThumbnailUrl
                                }}
                            />}
                    <CardImg
                        alt={name}
                        src={nearby.pictureThumbnailUrl}
                        className="card-img"
                        top
                        width="100%" />
                    <CardBody className="card-body">
                        <CardTitle tag="h5">
                            {name.charAt(0).toUpperCase() + name.slice(1)}
                        </CardTitle>
                    </CardBody>
                </Card>

            </div>
        )
    } else if (search) {
        return (
            <div className="card-animal">
                <Card
                    style={{
                        width: '18rem'
                    }}
                    onClick={handleClick}
                >
                    {favPets.includes(+id) ?
                        <Favorite
                            pet={{
                                petId: id,
                                petName: name,
                                petPic: search.pictureThumbnailUrl
                            }}
                            currFav={true} />
                        :
                        <Favorite
                            pet={{
                                petId: id,
                                petName: name,
                                petPic: search.pictureThumbnailUrl
                            }}
                        />}

                    <CardImg
                        alt={name}
                        src={search.pictureThumbnailUrl}
                        className="card-img"
                        top
                        width="100%" />
                    <CardBody className="card-body">
                        <CardTitle tag="h5">
                            {name.charAt(0).toUpperCase() + name.slice(1)}
                        </CardTitle>
                        <CardText className="card-text">
                            {search.ageGroup} - {search.breedPrimary}
                        </CardText>
                        {search.distance ?
                            <CardText className="card-text2">
                                {search.distance}  miles away
                            </CardText>
                            : <></>}

                    </CardBody>
                </Card>

            </div>
        )
    } else if (fav) {

        return (
            <div className="card-animal">
                <Card
                    style={{
                        width: '18rem'
                    }}
                    onClick={handleClick}
                >
                    <Favorite
                        id={fav.petId}
                        name={fav.petName}
                        pic={fav.petPic}
                        currFav={true} />

                    <CardImg
                        alt={fav.petName}
                        src={fav.petPic}
                        className="card-img"
                        top
                        width="100%" />
                    <CardBody className="card-body">
                        <CardTitle tag="h5">
                            {fav.petName}
                        </CardTitle>

                    </CardBody>
                </Card>

            </div>
        )
    }


}

export default AnimalCard 