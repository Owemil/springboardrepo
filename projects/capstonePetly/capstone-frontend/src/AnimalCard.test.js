import React from "react";
import PetlyContext from "./PetlyContext.js";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AnimalCard from "./AnimalCard.js";

const user = { favPets: [{ petId: 1 }, { petId: 2 }] }
const nearby = {
    name: "bubbles",
    pictureThumbnailUrl: "wwww.fakepic.com",
}
const search = {
    name: "tipsy",
    pictureThumbnailUrl: "wwww.fakepic.com",
    ageGroup: "young",
    breedPrimary: "pomeranian",
    distance: "2.66"
}
const fav = {
    petId: 2,
    petName: "dj",
    petPic: "wwww.fakepic.com"
}

const getAnimal = jest.fn()

it("should render without crashing (nearby)", function () {
    render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard nearby={nearby} id="2" />
        </PetlyContext.Provider>
    </MemoryRouter>)
})
it("should render without crashing (search)", function () {
    render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard search={search} id="3" />
        </PetlyContext.Provider>
    </MemoryRouter>)
})
it("should render without crashing (fav)", function () {
    render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard fav={fav} />
        </PetlyContext.Provider>
    </MemoryRouter>)
})

it("should match snapshot (nearby)", function () {
    const { asFragment } = render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard nearby={nearby} id="2" />
        </PetlyContext.Provider>
    </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (search)", function () {
    const { asFragment } = render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard search={search} id="3" />
        </PetlyContext.Provider>
    </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (fav)", function () {
    const { asFragment } = render(<MemoryRouter>
        <PetlyContext.Provider value={{
            user: user,
            getAnimal: getAnimal
        }}>
            <AnimalCard fav={fav} />
        </PetlyContext.Provider>
    </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
