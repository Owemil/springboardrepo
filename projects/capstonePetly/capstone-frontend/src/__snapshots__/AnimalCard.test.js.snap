// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot (fav) 1`] = `[Function]`;

exports[`should match snapshot (nearby) 1`] = `[Function]`;

exports[`should match snapshot (search) 1`] = `[Function]`;
